External/FEXCore/Source/Interface/Core/CPUID.cpp

```diff
@@ -826,14 +826,33 @@ FEXCore::CPUID::FunctionResults CPUIDEmu::Function_4000_0000h(uint32_t Leaf) {
   // CPUID documentation information:
   // 4000_0000h - 4FFF_FFFFh - No existing or future CPU will return information in this range
   // Reserved entirely for VMs to do whatever they want.
-  Res.eax = 0x40000000;
+  Res.eax = 0x40000001;
 
   // EBX, EDX, ECX become the hypervisor ID signature
   constexpr static char HypervisorID[12] = "FEXIFEXIEMU";
   memcpy(&Res.ebx, HypervisorID, sizeof(HypervisorID));
   return Res;
 }
 
+// Hypervisor CPUID information leaf
+FEXCore::CPUID::FunctionResults CPUIDEmu::Function_4000_0001h(uint32_t Leaf) {
+  FEXCore::CPUID::FunctionResults Res{};
+  if (Leaf == 0) {
+    // EAX[3:0] Is the host architecture that FEX is running under
+#ifdef _M_X86_64
+    // EAX[3:0] = 1 = x86_64 host architecture
+    Res.eax |= 0b0001;
+#elif defined(_M_ARM_64)
+    // EAX[3:0] = 2 = AArch64 host architecture
+    Res.eax |= 0b0010;
+#else
+    // EAX[3:0] = 0 = Unknown architecture
```

Comment:
In the case we do add support for more architectures, wouldn't it be better to have this path `#error` out instead of silently reporting an unknown architecture? (Also see my related comment in #1640.)